File: 3sem/4lab/Banks.Console/Handlers/Handler.cs
using Banks.Services;

namespace Banks.Console.Handlers;

public abstract class Handler : IHandler
{
    public IHandler? NextHandler { get; private set; } = null;

    public IHandler SetNext(IHandler nextHandler)
    {
        ArgumentNullException.ThrowIfNull(nextHandler);

        NextHandler = nextHandler;

        return nextHandler;
    }

    public abstract void Handle(string request, CentralBank centralBank);
}

File: 4sem/1lab/src/main/java/BanksRemake/Presentation/Handlers/Handler.java
package banksRemake.presentation.handlers;

import banksRemake.domain.exceptions.InvalidBankAccountOperationException;
import banksRemake.domain.exceptions.InvalidBankOperationException;
import banksRemake.domain.exceptions.InvalidDepositInterestRateSelectorException;
import banksRemake.domain.exceptions.InvalidMoneyGapException;
import banksRemake.domain.services.CentralBank;

import java.util.Scanner;

public interface Handler {
    Handler getNext();
    Handler setNext(Handler next);

    void handle(String request, CentralBank centralBank, Scanner scanner) throws InvalidBankOperationException, InvalidBankAccountOperationException, InvalidDepositInterestRateSelectorException, InvalidMoneyGapException;
}
